posixlib/src/main/scala/scala/scalanative/posix/dirent.scala

```diff
@@ -1,42 +1,127 @@
 package scala.scalanative
 package posix
 
+import scalanative.posix.sys.types.ino_t
+
 import scala.scalanative.unsafe._, Nat._
 
-@extern
+object DirentTypes {
+  type _256 = Digit3[_2, _5, _6] // see comment above 'type dirent' below.
+
+  type DIR = CStruct0 // An opaque structure from os. Deconstruct a your peril.
```

Comment:
I think these types should be publically available only via the `object dirent`. We should make `DirentTypes` private

Reply:
Agreed. Thank you for review.

I am still chasing a 32 bit problem. So you may see some weirdness in the build logs for awhile.